x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -73,28 +74,37 @@
 
     public static final TimeValue MIN_SNAPSHOT_CACHE_SYNC_INTERVAL = TimeValue.timeValueSeconds(1L);
     public static final Setting<TimeValue> SNAPSHOT_CACHE_SYNC_INTERVAL_SETTING = Setting.timeSetting(
-        SETTINGS_PREFIX + "sync_interval",
+        SETTINGS_PREFIX + "sync.interval",
         TimeValue.timeValueSeconds(60L),                        // default
         MIN_SNAPSHOT_CACHE_SYNC_INTERVAL,                       // min
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
 
     public static final Setting<Integer> SNAPSHOT_CACHE_MAX_FILES_TO_SYNC_AT_ONCE_SETTING = Setting.intSetting(
-        SETTINGS_PREFIX + "max_files_to_sync",
+        SETTINGS_PREFIX + "sync.max_files",
         10_000,                                                 // default
         0,                                                      // min
-        Integer.MAX_VALUE,                                      // min
+        Integer.MAX_VALUE,                                      // max
         Setting.Property.NodeScope,
         Setting.Property.Dynamic
     );
 
+    public static final Setting<TimeValue> SNAPSHOT_CACHE_SYNC_SHUTDOWN_TIMEOUT = Setting.timeSetting(
+        SETTINGS_PREFIX + "sync.shutdown_timeout",
+        TimeValue.timeValueSeconds(60L),                        // default
```

Comment:
Perhaps just 10 seconds default, since we only need to wait for one fsync and if it takes more than 10s to do one, we really want to continue shutting down the node anyway? The other `doStop` timeouts that I found (did not search thoroughly though) are in the 10-30s range.

Reply:
Agreed.